src/main/cluster.ts

```diff
@@ -201,6 +202,15 @@ export class Cluster implements ClusterModel, ClusterState {
    * @observable
    */
   @observable allowedNamespaces: string[] = [];
+
+  /**
+   * Is true if `allowedNamespaces` was filled by a successful kubeAPI list
+   * namespaces request.
+   *
+   * @observable
+   */
+  @observable listedNamespaces = false;
```

Comment:
Why do we need this when we have already `cluster.accessibleNamespaces` and `cluster.allowedNamespaces`?
If `accessibleNamespaces` is provided by user it must be used first.

Otherwise maybe better to use this as I understand it correct it serves same purpose?

```ts
  @computed get listedNamespaces(): boolean {
    return !this.accessibleNamespaces.length;
  }
```

Reply:
Because `allowedNamespaces` might also be from the context. See line 700

Reply:
I tought `allowedNamespaces` _must_ be _only_ from the context? No?

Reply:
Or by listing via the kube api ( which I believe is the most common)

Reply:
Also there are some inconsistency between 2 methods imho:

```ts
  protected async getAllowedNamespaces() { // get allowed?
    if (this.accessibleNamespaces.length) {
      return this.accessibleNamespaces; // No, get accessible!
    }

  protected async getAllowedResources() {
    try {
      if (!this.allowedNamespaces.length) { // why not checking accessible as well?
        return [];
      }
```

Reply:
Agreed, but if `getAllowedResources` is only ever called after `getAllowedNamespaces` it "just works"

Reply:
There are *lots* of things I would like to change to `cluster.ts` and the above it part of it but I don't think that it is in scope for this PR.

Reply:
I don't really like the idea of yet-another-public-observable because it will go to the extension API and is really hard to take away later. Any possibility that we could resolve this from the existing information (as we did before)?